### PR TITLE
2FA support for OAuth & personal tokens

### PR DESCRIPTION
**Pull request by Betree**
Part of https://github.com/opencollective/opencollective/issues/6088

- [x] Add a special flag on Applications and personal tokens to allow 2FA bypass
- [x] Bypass 2FA check when this flag is set
- [x] Add a warning (or error) when using a 2FA code with OAuth (apart from the 1kproject script)

### CODE REVIEW

File: server/graphql/v2/mutation/PersonalTokenMutations.ts [hunk 1]
```diff
@@ -74,9 +74,19 @@ const updatePersonalToken = {
...
     } else if (!req.remoteUser.isAdminOfCollective(personalToken.collective)) {
       throw new Forbidden('Authenticated user is not the token owner.');
+    } else if (
+      !isUndefined(args.personalToken.preAuthorize2FA) &&
+      args.personalToken.preAuthorize2FA !== personalToken.preAuthorize2FA &&
+      req.personalToken
+    ) {
+      throw new Error(
+        'You cannot change the preAuthorize2FA value of the token you are using. Please use the interface to change it.',
+      );
```
Comment (znarf) [inline server/graphql/v2/mutation/PersonalTokenMutations.ts:84]: Why not something more generic like: no update to a personalToken when authenticating with a personalToken?

File: server/graphql/v2/mutation/PersonalTokenMutations.ts [hunk 2]
```diff
@@ -74,9 +74,19 @@ const updatePersonalToken = {
...
     }
 
-    const updateParams = pick(args.personalToken, ['name', 'scope', 'expiresAt']);
+    const hasCriticalChanges = ['scope', 'expiresAt'].some(field => !isUndefined(args.personalToken[field]));
```
Comment (znarf) [inline server/graphql/v2/mutation/PersonalTokenMutations.ts:87]: preAuthorize2FA is not a critical change?

**Review by Betree — COMMENTED** [review]
Made the changes and added some tests

**Review by znarf — APPROVED** [review]
Looks good!

**Pull request merged**